### PR TITLE
File cleanup

### PR DESCRIPTION
**Pull request by egthomas**
This pull request removes the VT-specific IDL library and compare_files.py script described in issue #6 .  Also strips out and ignores nearly all of the .h files in the include/ directory (except mpfit.h) which are populated there during compilation.  Finally, the install notes have been simplified as it is not necessary to copy the cdf.h file if the CDF_PATH environment variable is properly set.

### CODE REVIEW

**Comment by ksterne** [comment]
I've added back the fitexfilter binary that was wiped out.  I had forgotten past discussions that we need to keep this around as people (at least here at VT) are using this binary, but we do not have the source code to recompile it.  So, while it may become a useless binary at some point, it's still useful today.  Hopefully one day someone will re-write the code or write a better filter and keep the source code around.

**Comment by egthomas** [comment]
I have to disagree here - I don't think that a binary which was compiled years ago on a particular machine and doesn't have any supporting source code or documentation should be included in the main RST distribution.  Can you guys put it in a different directory on your local setup so that you can still run it as needed?

**Comment by egthomas** [comment]
Although I still disagree I've left the fitexfilter in the commit history for now to avoid confusion.  This latest round of commits removes and stops tracking the dlm/ and doc/ directories which are populated during the compilation procedure by executing `make.code superdarn rst` and `make.doc superdarn rst` respectively.  

This is also where I'm going to draw the line on adding commits to this pull request.  If someone wants to try compiling this branch to verify there are no errors and everything is still populated it would be greatly appreciated.

**Comment by ksterne** [comment]
@egthomas, I can certainly agree with the dangers of having a binary which we can't see the source code for in addition to why all the other pre-compiled files were removed.  It seems as though I took the fitexfilter binary out and placed it in our local davit binary directory and at least in shell its pointing to the davit directory.

The pro I see to having the fitexfilter up here is that then we are not the sole source of the binary.  That also means no one aside from VT has access to it.

I don't have strong feelings either way.  I worry about losing this if we're the sole source but I also understand the frustration and complications of having a binary we don't have source code for. Maybe we should get a ruling from the DAWG chair?  @pasha-ponomarenko?

**Comment by sshepherd** [comment]
The issue here is that this is a binary that probably only works on
specific VT hardware and will likely stop working when any of the dependent
libraries change. Without source code the lifetime of this piece of code is
limited. It doesn't make any sense to keep binaries in the RST distribution.

**Comment by egthomas** [comment]
From a file backup perspective the fitexfilter binary is still contained in at least one branch of the VTRST3.5 repository.

**Comment by pasha-ponomarenko** [comment]
Kevin S, could you please clarify the exact function of this binary?

**Comment by ksterne** [comment]
From the --help output:

This program takes a fitex (fitacf) format file and performs a boxcar filter
calling format:
fitexfilter [-options] fitex_file > filtered_fitex_file
options:
-help : display this menu
thresh f.f : filter threshold, a float in [0,1], default is 0.4
tdim x : time dimension of boxcar filter, 1 or 3, default is 3
bdim x : beam dimension of boxcar filter, 1 or 3, default is 3
rdim x : range dimension of boxcar filter, 1 or 3, default is 3
NB -- a 1x1x1 filter is equivalent to no filter at all
nocomb : do not combine multiple camping beams for a single scan
NB -- nocomb uses a 3x1x3 (TxBxR) filter on camping beams
vb : verbose

So, in short this is a boxcar filter for fit-level data.  

Again, it seems as though I've been able to pull this binary out of the rst and its working for our immediate uses here at VT.  I do take that, for now, the VTRST3.5 repo can serve as a backup...though I think it's now only in one branch.

Thoughts @pasha-ponomarenko?

**Comment by egthomas** [comment]
Hang on, I guess I had assumed fitexfilter was AJ's ionospheric/ground scatter search algorithm.  Instead it sounds like it does the same thing (boxcar median filter) as the fitfilter function in codebase/superdarn/src.bin/tk/tool.

**Comment by ksterne** [comment]
Yea, no, this isn't the blob analysis software that AJ also developed.  The result of the fitexfilter binary can be seen with:

http://vt.superdarn.org/tiki-index.php?page=Latest+Data

by toggling the filter on/off by clicking on the "Turn Filter On/Off" link.  Right now I think the filtered plot is set to not mark groundscatter.

**Comment by pasha-ponomarenko** [comment]
Well, I don't see why would we need to distritbute it. It is very easy to replicate it in IDL or Python if necessary anyway... So, I think we can safely remove it from here.

**Comment by egthomas** [comment]
OK, fitexfilter has been removed.  Getting back to the original intention of this pull request, it should now address issues #4 (removing compiler generated files) and #6 (removing VT-specific IDL software).  As a result a zipped download copy of the RST will be reduced in size by nearly 30% (88MB vs 124MB).

There may be more files to trim from the ROS side but I'm not as familiar with that portion of the RST and am hesitant to attempt a similar purge.  I would suggest merging this sooner rather than later so we all have cleaner file tracking in place as we continue to modify the software.

**Comment by ksterne** [comment]
Sounds good here.  I don't know that much about the fitexfilter, but like we've said, I've got some backups if we need.

I have confirmed (not from scratch) that this branch does compile without any problems.  Should we wait on double confirmation or is the developer pool too small for that?  If it isn't, I can merge (pull requesters shouldn't merge their own branches).

**Comment by egthomas** [comment]
The developer pool may be too small like you say but at this point I don't see the harm in waiting until Monday to give a little extra time.
